### PR TITLE
Feature/post requests

### LINKED ISSUE #13: Update Page on Post Requests
- [x] Sleep Data
- [x] {"userID": integer, "date": string, "hoursSlept": integer, "sleepQuality": integer}
- [x] Activity Data
- [x] {"userID": integer, "date": string, "numSteps": integer, "minutesActive": integer, "flightsOfStairs": integer}
- [x] Hydration Data
- [x] {"userID": integer, "date": string, "numOunces": integer}
- [x] Do proper error handling for your users to ensure that they are getting data and are submitting their POST requests successfully. An example is handling the case where they submit their data and an error message returns from the request. Also validate the input fields on the client-side.

### LINKED ISSUE #34: GET Request on Page Load
When the page loads, the user GET request should be invoked

### PR DESCRIPTION
**Pull request by alexthompson207**
### Description
* Is this a feature or a fix?

This is a feature.

* Where should the reviewer start?

Start in the `scripts.js` file, look at the function `handlePostRequest(link, body)` then in the `API.js` file look at the `addNewData(link, newData)` method which is a dynamic POST request method.

* What functionalities do these changes effect? Why were these changes made? What was the motivation behind these changes?

Changes were described above ^. User is able to input hydration, sleep, and activity information and their new information is displayed on the DOM.

* What outside features or research did you use to implement this fix?

Lessons.

* How should this be tested?

Load the website locally and try to input new activity/sleep/hydration information, page should update.

### Relevant Screen Shots
***
#### Please review considerations below:
- [x] Follows Turing Style Guidelines
- [x] Changes generate no new warnings or errors
- [x] Checked code for and corrected any spelling errors
- [ ] README has been updated(if applicable)

### CODE REVIEW

File: src/API.js [hunk 1]
```diff
@@ -7,25 +6,11 @@ const fetchAPIData = {
       .then(data => data.userData)
       .catch(err => console.log(err))
     userSet = Promise.resolve(userData).then(data => {
-      // console.log(userSet);
       return data;
     })
     return userSet
   },
```
Comment (aemiers) [inline src/API.js:13]: Great job cleaning up this file! Looks fantastic!

File: src/Calculation.js [hunk 2]
```diff
@@ -12,7 +12,12 @@ class Calculation {
 
   calculateDailyData(id, date, property) {
     const findDataByDate = this.dataSet.find(data => id === data.userID && date === data.date);
-    return findDataByDate[property];
+    if (findDataByDate && findDataByDate[property]) {
+      return findDataByDate[property];
+    }
+    else {
+      return 0;
```
Comment (aemiers) [inline src/Calculation.js:19]: Nice error handling!

File: src/scripts.js [hunk 3]
```diff
@@ -114,46 +117,51 @@ function hideAllForms() {
...
+  enteredActivityInfo.flightsOfStairs = parseInt(formActivityFlights.value);
+  checkForCompletion("http://localhost:3001/api/v1/activity", enteredActivityInfo);
 }
 
-function checkForCompletion(composedObject) {
+function checkForCompletion(url, composedObject) {
   const values = Object.values(composedObject);
-  console.log('check for completion values', values)
-  if (values.includes("")) {
+  if (values.includes("" || NaN)) {
```
Comment (aemiers) [inline src/scripts.js:159]: NaN! Really stumped me for a second there! Great job!

**Review by itsnameissteven — APPROVED** [review]
Great work. Love the addition of sad paths

**Pull request merged**